src/main/kotlin/com/xenomachina/argparser/ArgParser.kt

```diff
@@ -458,6 +458,15 @@ class ArgParser(args: Array<out String>,
         finished = true
     }
 
+    /**
+     * Provides an instance of T, where all arguments have already been parsed and validated
+     */
+    inline fun <reified T> provideParsed(): T {
```

Comment:
I'm not a big fan of using `getDeclaredConstructor` or `newInstance`, to be honest. I'd rather take a function than a class, as it's both more flexible, and more type safe. (This implementation fails at runtime with a `NoSuchMethodException` if given a class lacking the right type of constructor.)

What do you think about changing it to something like this?:
```
inline fun <T> provideParsed(constructor: (ArgParser) -> T): T {
    val provided = constructor(this)
    force()
    return provided
}
```

This changes the way you call it slightly:
```
parser.provideParsed<Args>()
```
becomes:
```
parser.provideParsed(::Args)
```
but it also lets you pass additional parameters to the Args constructor, or potentially even choose which args class to use based on things like environment variables.
```
parser.provideParsed { parser -> /* code that creates Args */ }
```

Perhaps the name "provideParsed" doesn't work for this different signature. Maybe `parse` or `parseInto` work better with that signature.

I think it'd also make sense to restrict this method to only be used on a "clean" parser. You can add something like this to the end of `ArgParser`:
```
val builtinDelegateCount = delegates.size
```
And then at the beginning of this method add something like:
```
if (delegates.size > builtinDelegateCount) {
    throw IllegalStateException("parseSubCommand can only be used with a clean ArgParser")
}
```

Interestingly, I'm adding a similar method in a branch where I'm working on adding sub-command support. Right now that method's signature looks like:

```
fun <C, S> parseSubCommand(
            commonOptionsConstructor: (OptionParser) -> C,
            subCommandMap: Map<String, (SubCommandArgs<C>) -> S>) : S
```

I'll change the name of that method to be consistent with this one.

Reply:
I like the name `parseInto`. I'll make these changes in the next day or so.

Reply:
Hello. Have you had a chance to finish this up? 🙂

Reply:
Hey, sorry, I got really busy at work. It's pretty much done, just fighting a failing test.

Reply:
@xenomachina Pushed the requested changes, finally :)